File: 프로그래머스/Level1/시저 암호.py
# https://programmers.co.kr/learn/courses/30/lessons/12926

def solution(s, n):
    answer = ''
    for a in s:
        if a == ' ':
            answer += a
            continue
        temp = ord(a)
        if 65 <= temp <= 90:
            temp += n
            if 65 <= temp <= 90:
                answer += chr(temp)
            else:
                answer += chr(temp - 26)
            continue
        elif 97 <= temp <= 122:
            temp += n
            if 97 <= temp <= 122:
                answer += chr(temp)
            else:
                answer += chr(temp - 26)
    return answer

File: 프로그래머스/Level1/시저 암호.js
// https://programmers.co.kr/learn/courses/30/lessons/12926

function solution(s, n) {
	let answer = '';
	for (let i = 0; i < s.length; i++) {
		if (s[i] === ' ') answer += ' ';
		else {
			let temp;
			if (s.charCodeAt(i) >= 65 && s.charCodeAt(i) <= 90)
				temp = s.charCodeAt(i) + n <= 90 ? s.charCodeAt(i) + n : s.charCodeAt(i) + n - 26;
			else temp = s.charCodeAt(i) + n <= 122 ? s.charCodeAt(i) + n : s.charCodeAt(i) + n - 26;
			answer += String.fromCharCode(temp);
		}
	}
	return answer;
}
